spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -265,9 +268,27 @@ private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobPar
 					}
 				}
 				if (!existsFlag) {
-					result.add(String.format("%s(%s)=%s", key,
+					String param;
+					if (BOOT3_VERSION.equals(schemaTarget)) {
+						if (JobParameter.ParameterType.LONG.equals(jobParametersMap.get(key).getType())) {
```

Comment:
I assume by this time that `jobParametersMap.get(key)` is always non-null?

Reply:
Yes.  At line jobParametersMap is a non modifiable map and would have thrown an NPE above this.   Good question